pkg/virt-handler/node-labeller/node_labeller.go

```diff
@@ -230,20 +226,7 @@ func (n *NodeLabeller) patchNode(originalNode, node *v1.Node) error {
 		})
 	}
 
-	if !equality.Semantic.DeepEqual(originalNode.Annotations, node.Annotations) {
```

Comment:
@RamLavi Are you sure its completely safe to remove the annotation handling? what about `[LabellerSkipNodeAnnotation](https://github.com/kubevirt/kubevirt/blob/f24a4618b3d2fc4db7e5db38b35d2fccc6d11e96/staging/src/kubevirt.io/api/core/v1/types.go#L928)`

Reply:
WRT the [LabellerSkipNodeAnnotation](https://github.com/kubevirt/kubevirt/blob/f24a4618b3d2fc4db7e5db38b35d2fccc6d11e96/staging/src/kubevirt.io/api/core/v1/types.go#L928) annotation, AFAIU this annotation is set by the user in order to opt-out the node-labeling entirely (see [here](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-handler/node-labeller/node_labeller.go#L214)). If that annotation is set then node-labeller will not patch anything, so this part of "patchNode" code won't run. In other words, I'm not removing this annotation's functionality.

As for if I'm completely sure that the annotations are not needed, it says so on the commit message that introduced the node-labeller on release 0.40 (see commit [here](https://github.com/kubevirt/kubevirt/pull/4916/commits/5eb88073d8675657dd0b0b4f998e2f99663150d1#)):
```
This commit migrates the node labeller from ssp operator into kubevirt.
This feature labels nodes with host supported cpu models and features and hyper-v features.

After new labeller runs it deletes all old labels / annotations and
creates new labels with new namespace.
```
So IIUC now that we don't have to support the upgrade from this version then we can remove this code handling old annotations/labels.


... But I didn't stop there and further examined the code to double check.. 
Notice that:
- The original (now archived) node-labeller repo added both labels and annotations (see [here](https://github.com/kubevirt/node-labeller/blob/4fbba47ae442cc14ac7eda00b8b7b2353e72dee3/pkg/node/node.go#L52)).
- When the node-labeller moved to kubevirt ([see](https://github.com/kubevirt/kubevirt/pull/4916) main branch PR) on v0.40.0 - the behavior changed: 
   - new labels were only added to the metadata.labels (see [here](https://github.com/kubevirt/kubevirt/pull/4916/commits/5eb88073d8675657dd0b0b4f998e2f99663150d1#diff-06e4ea6b40733549df9ae0810c7e4aa3e90e24d10a4c78e49c40c2254153e3baR242)).
   - annotations were only used in order to be removed (see [here](https://github.com/kubevirt/kubevirt/pull/4916/commits/5eb88073d8675657dd0b0b4f998e2f99663150d1#diff-06e4ea6b40733549df9ae0810c7e4aa3e90e24d10a4c78e49c40c2254153e3baR260))
   - Specifically - only the annotations with the DeprecatedLabellerNamespaceAnnotation prefix were removed. This deprecated annotation is exactly the one we are now no longer supporting. 

Do you think I should check any further?

Reply:
@RamLavi Thank you for the detailed explanation!